modules/rules_java/4.0.0/MODULE.bazel

```diff
@@ -0,0 +1,10 @@
+module(
+    name = "rules_java",
+    version = "4.0.0",
+    compatibility_level = 1,
+)
+bazel_dep(name = "bazel_skylib", version = "1.0.3")
+
```

Comment:
just curious, what's our policy for adding the MODULE.bazel file itself as a patch? I see you have done it for some other modules (rules_python for example), but not done it here. It probably doesn't matter too much either way; I just thought we should be consistent.